clients/client-java/src/main/java/com/datastrato/gravitino/client/BaseCatalogDTO.java

```diff
@@ -0,0 +1,232 @@
+/*
+ * Copyright 2024 Datastrato Pvt Ltd.
+ * This software is licensed under the Apache License version 2.
+ */
+package com.datastrato.gravitino.client;
+
+import com.datastrato.gravitino.NameIdentifier;
+import com.datastrato.gravitino.Namespace;
+import com.datastrato.gravitino.dto.AuditDTO;
+import com.datastrato.gravitino.dto.CatalogDTO;
+import com.datastrato.gravitino.dto.requests.SchemaCreateRequest;
+import com.datastrato.gravitino.dto.requests.SchemaUpdateRequest;
+import com.datastrato.gravitino.dto.requests.SchemaUpdatesRequest;
+import com.datastrato.gravitino.dto.responses.DropResponse;
+import com.datastrato.gravitino.dto.responses.EntityListResponse;
+import com.datastrato.gravitino.dto.responses.SchemaResponse;
+import com.datastrato.gravitino.exceptions.NoSuchCatalogException;
+import com.datastrato.gravitino.exceptions.NoSuchSchemaException;
+import com.datastrato.gravitino.exceptions.NonEmptySchemaException;
+import com.datastrato.gravitino.exceptions.SchemaAlreadyExistsException;
+import com.datastrato.gravitino.rel.Schema;
+import com.datastrato.gravitino.rel.SchemaChange;
+import com.datastrato.gravitino.rel.SupportsSchemas;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Preconditions;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * BaseCatalogDTO is the base class for all the catalog DTOs. It provides the common methods for
+ * managing schemas in a catalog.
+ */
+public class BaseCatalogDTO extends CatalogDTO implements SupportsSchemas {
```

Comment:
We should expose the class `xxxDTO`. The class DTO is only used for transportation between the client and the server.

Reply:
How about renaming it to BaseCatalog?

Reply:
I prefer SchemaCatalog for this class. Maybe we can heard from other people's suggestion.

Reply:
I rename it to `GravitinoCatalog`, it's similar to `GravitinoMetaLake`. What do you think?
```
public class GravitinoMetaLake extends MetalakeDTO implements SupportsCatalogs
```

```
public class GravitinoCatalog extends CatalogDTO implements SupportsSchemas
```

Reply:
OK for me. This seems better than BaseCatalog.

Reply:
Actually, I have some concern about the name. If we use this class, it means that the other catalogs should also have the concept of schema in the future.

Reply:
If all catalogs have the concept of schema, what's the disadvantage?

Reply:
> If all catalogs have the concept of schema, what's the disadvantage?

I worry whether all catalogs need it. If some catalogs have already mature concepts, we add a concept of schema, it may be weird for the catalog.

Reply:
> > If all catalogs have the concept of schema, what's the disadvantage?
> 
> I worry whether all catalogs need it. If some catalogs have already mature concepts, we add a concept of schema, it may be weird for the catalog.

It sounds meaningful. I'm not very sure about this. Do we need to have our own standards？ If we have a schema, It's convenient to split the resource. If some resource doesn't need a schema, it can use the same schema by default.

Reply:
@jerryshao @qqqttt123 What about renaming it to `SchemaCatalog`?

Reply:
> @jerryshao @qqqttt123 What about renaming it to `SchemaCatalog`?

I'm ok.

Reply:
@coolderli sorry for late response. I'm busy on 0.4.0 release. I will give an detailed review today and back to you the  feedbacks.